src/stdatm/speed_parameters.py

```diff
@@ -10,3 +11,264 @@
+def _compute_subsonic_impact_pressure(mach, pressure):
+    return pressure * ((1.0 + 0.2 * mach**2) ** 3.5 - 1.0)
+
+
+def _compute_supersonic_impact_pressure(mach, pressure):
+    # Rayleigh law
+    # https://en.wikipedia.org/wiki/Rayleigh_flow#Additional_Rayleigh_Flow_Relations
+    return pressure * (166.92158 * mach**7 / (7 * mach**2 - 1.0) ** 2.5 - 1.0)
+
+
+@singledispatch
+def compute_impact_pressure(mach, pressure):
+    """
+
+    :param mach: no unit
+    :param pressure: in Pa
+    :return: impact pressure in Pa
+    """
+    # Implementation for numpy arrays
+    mach = np.asarray(mach)
+    idx_subsonic = mach <= 1.0
+    idx_supersonic = np.logical_not(idx_subsonic)
+
+    if np.shape(pressure) != np.shape(mach):
+        pressure = np.broadcast_to(pressure, np.shape(mach))
+    else:
+        pressure = np.asarray(pressure)
+
+    impact_pressure = np.empty_like(mach)
+    impact_pressure[idx_subsonic] = _compute_subsonic_impact_pressure(
+        mach[idx_subsonic], pressure[idx_subsonic]
+    )
+    impact_pressure[idx_supersonic] = _compute_supersonic_impact_pressure(
+        mach[idx_supersonic], pressure[idx_supersonic]
+    )
+    return impact_pressure
+
+
+@compute_impact_pressure.register
+def _(mach: Number, pressure: Number):
+    # Implementation for floats
+    if mach <= 1.0:
+        return _compute_subsonic_impact_pressure(mach, pressure)
+    else:
+        return _compute_supersonic_impact_pressure(mach, pressure)
+
+
+# CALIBRATED AIRSPEED =================================================================
+#         Computation is done using Eq. 3.16 and 3.17 from:
+#         Gracey, William (1980), "Measurement of Aircraft Speed and Altitude",
+#         NASA Reference Publication 1046.
+#         https://apps.dtic.mil/sti/pdfs/ADA280006.pdf
+
+# Pre-calculation of some equation constants for sake of speed.
+GAMMA_MINUS_ONE_OVER_GAMMA = (1.4 - 1.0) / 1.4
+GAMMA_MINUS_ONE_OVER_TWO_GAMMA = (1.4 - 1.0) / 2.8
+COEFF_1 = 6.0**2.5 * 1.2**3.5
+
+
+def _compute_cas_low_speed(impact_pressure, sea_level_pressure, sea_level_speed_of_sound):
+    return sea_level_speed_of_sound * np.sqrt(
+        5.0 * ((impact_pressure / sea_level_pressure + 1.0) ** GAMMA_MINUS_ONE_OVER_GAMMA - 1.0)
+    )
+
+
+def _equation_cas_high_speed(cas, impact_pressure, sea_level_pressure, sea_level_speed_of_sound):
+    return (
+        cas
+        - sea_level_speed_of_sound
+        * (
+            (impact_pressure / sea_level_pressure + 1.0)
+            * (7.0 * (cas / sea_level_speed_of_sound) ** 2 - 1.0) ** 2.5
+            / COEFF_1
+        )
+        ** GAMMA_MINUS_ONE_OVER_TWO_GAMMA
+    )
+
+
+def _compute_cas_high_speed(impact_pressure, sea_level_pressure, sea_level_speed_of_sound):
+    root = fsolve(
+        _equation_cas_high_speed,
+        x0=sea_level_speed_of_sound * np.ones_like(impact_pressure),
+        args=(impact_pressure, sea_level_pressure, sea_level_speed_of_sound),
+    )
+    return root
```

Comment:
The `root` function from scipy.optimize (https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.root.html#scipy.optimize.root) does the same but it appears it is designed for vecotr function. It may be worth to use the singledispatch decorator here as well and use either fsolve or root.

Reply:
I realize the actual problem is to use `fsolve`, which is a legacy function that is not recommended anymore. Your point still stands regarding `root` and `root_scalar`. I'll change that.

Reply:
Done in 3c842325333e03c4dcf678ca9897f3197f65bce1 and 0ede429fa4b38a7d1616ca0698de68cbfa8c1551.